tesseract_time_parameterization/src/time_optimal_trajectory_generation.cpp

```diff
@@ -957,6 +1073,92 @@ bool Trajectory::isValid() const { return valid_; }
 
 double Trajectory::getDuration() const { return trajectory_.back().time_; }
 
+bool Trajectory::assignData(TrajectoryContainer& trajectory, double path_tolerance) const
+{
+  std::list<Trajectory::TrajectoryStep>::const_iterator it = trajectory_.begin();
+  long cnt = 0;
+  for (long i = 0; i < trajectory.size(); ++i)
+  {
+    const Eigen::VectorXd& p = trajectory.getPosition(i);
+    for (; it != trajectory_.end(); ++it)
+    {
+      Eigen::VectorXd compare_p = getPosition(it->time_).head(trajectory.dof());
+      if (!((p - compare_p).norm() > path_tolerance))
```

Comment:
Yeah this is what I did originally as well. However, when multiple points were within the tolerance, it assigns most of them to the next composite. That's why I started checking the error and waiting until the error stopped going down before incrementing.

Reply:
See here if you are interested https://github.com/ros-industrial-consortium/tesseract_planning/pull/30/commits/8407043606bf6a789b7bf88b4e5e6c724cb39f67#diff-35b696ce9c09c51a59cc90167b5987ccb88660c8e4b03054a2f2d594e9a7e137R194

Reply:
I have added in your logic to improve finding the correct point on the trajectory.